sycl/source/detail/pi_opencl.cpp

```diff
@@ -78,6 +78,43 @@ pi_result OCL(piextDeviceSelectBinary)(
   return PI_SUCCESS;
 }
 
+pi_result OCL(piQueueCreate)(pi_context context, pi_device device,
+                             cl_command_queue_properties properties,
+                             pi_queue *queue) {
+  size_t devVerSize;
+  cl_int ret_err;
+  ret_err = clGetDeviceInfo(pi_cast<cl_device_id>(device), CL_DEVICE_VERSION, 0,
+                            NULL, &devVerSize);
+
+  if (ret_err != CL_SUCCESS)
+    return pi_cast<pi_result>(ret_err);
+
+  std::string devVer(devVerSize, '\0');
+  clGetDeviceInfo(pi_cast<cl_device_id>(device), CL_DEVICE_VERSION, devVerSize,
+                  &devVer.front(), NULL);
+
+  if (ret_err != CL_SUCCESS)
+    return pi_cast<pi_result>(ret_err);
+
+  if (devVer.find("OpenCL 1.0") != std::string::npos ||
+      devVer.find("OpenCL 1.1") != std::string::npos ||
+      devVer.find("OpenCL 1.2") != std::string::npos) {
+    *queue = pi_cast<pi_queue>(clCreateCommandQueue(
```

Comment:
please assert queue is not null (PI_ASSERT)

Reply:
@smaslov-intel Do you want to consider the case of a conforming `clCreateCommandQueue` to fail without setting up a correct `ret_err`?
For now it looks like there is no assertion on the other OpenCL calls...

Reply:
The clCreateCommandQueue will not fail, but PI will.
Instead of asserting we can choose to return error code from piQueueCreate in case no queue pointer was provided, but that does not seem useful to me.

Reply:
added an PI_ASSERT outside of the If statement a line or so above to check if the passed in queue is null, hopefully what you intended.